### PR TITLE
Static methods on custom class/struct

### PR DESCRIPTION
**Pull request by Roms1383**
Support static methods on custom class/struct.

### CODE REVIEW

**Comment by Roms1383** [comment]
```rs
ClassExport::<MyStruct>::builder()
.static_methods(static_methods![c"DoSomething" => MyStruct::do_something])
.build(),
```
```swift
public native struct MyStruct {
    public native static func DoSomething(self: MyStruct, value: Float) -> Void;
}

/// Game.TestMe();
public static exec func TestMe(game: GameInstance) {
    let o = new MyStruct();
    o.DoSomething(2077.0);
}
```

```rs
#[derive(Debug, Default, Clone)]
#[repr(C)]
pub struct MyStruct;

unsafe impl NativeRepr for MyStruct{
    const NAME: &'static str = "MyStruct";
}
/// required by:
/// the function or associated item `builder` exists for struct `ClassExport<MyStruct>`,
/// but its trait bounds were not satisfied the following trait bounds were not satisfied:
/// `audioware::MyStruct: ScriptClass`
unsafe impl ScriptClass for MyStruct{
    type Kind = Native;
    const NAME: &'static str = "MyStruct";
}

impl MyStruct{
    pub fn do_something(self, value: f32) {
        crate::utils::info(format!("hello from MyStruct::do_something({value})"));
    }
}
```

File: src/types/rtti.rs [hunk 1]
```diff
@@ -1088,12 +1088,16 @@ impl Drop for Method {
...
         full_name: &CStr,
         short_name: &CStr,
         class: &Class,
         handler: FunctionHandler<IScriptable, R>,
-    ) -> PoolRef<Self> {
+        flags: FunctionFlags,
+    ) -> PoolRef<Self>
+    where
+        C: ScriptClass,
+    {
```
Comment (jac3km4) [inline src/types/rtti.rs:1100]:
```suggestion
    where
        C: ScriptClass,
    {
```
the C type is not used except for this bound, it should be removed

File: src/invocable.rs [hunk 2]
```diff
@@ -213,6 +213,22 @@ impl GlobalMetadata {
...
+    /// Converts this metadata into a [`StaticMethod`] instance, which can be registered with
+    /// [RttiSystemMut](crate::RttiSystemMut).
+    pub fn to_rtti_static_method<C: ScriptClass>(&self) -> PoolRef<StaticMethod> {
+        let mut flags = FunctionFlags::default();
+        flags.set_is_native(true);
+        flags.set_is_final(true);
+        flags.set_is_static(true);
+
+        let rtti = RttiSystem::get();
+        let class = rtti.get_class(CName::new(C::NAME)).expect(C::NAME);
```
Comment (jac3km4) [inline src/invocable.rs:226]:
```suggestion
        let class = rtti.get_class(CName::new(C::NAME)).expect("class should exist");
```

Comment (Roms1383) [reply src/invocable.rs]: done in 95ffdf4

File: src/types/rtti.rs (hunk 1 above)
Comment (Roms1383) [reply src/types/rtti.rs]: done in 95ffdf4

File: src/types/rtti.rs [hunk 3]
```diff
@@ -1088,11 +1088,12 @@ impl Drop for Method {
 pub struct StaticMethod(red::CClassStaticFunction);
 
 impl StaticMethod {
-    pub fn new<R>(
+    pub fn new<C, R>(
```
Comment (jac3km4) [inline src/types/rtti.rs:1091]:
```suggestion
    pub fn new<R>(
```

Comment (Roms1383) [reply src/types/rtti.rs]: Removed in 45b04a8

File: src/invocable.rs [hunk 4]
```diff
@@ -213,6 +213,24 @@ impl GlobalMetadata {
...
+        flags.set_is_native(true);
+        flags.set_is_final(true);
+        flags.set_is_static(true);
+
+        let rtti = RttiSystem::get();
+        let class = rtti
+            .get_class(CName::new(C::NAME))
+            .expect("class should exist");
+
+        let mut func = StaticMethod::new::<_>(self.name, self.name, class, self.func, flags);
```
Comment (jac3km4) [inline src/invocable.rs:230]:
```suggestion
        let mut func = StaticMethod::new(self.name, self.name, class, self.func, flags);
```
small nitpick

**Pull request merged**